yourbench/pipeline/answer_generation.py

```diff
@@ -0,0 +1,276 @@
+            calls.append(InferenceCall(messages=[user_msg], tags=["zero_shot"]))
+            row_map.append(i)
+
+        elif scenario == "with_correct_chunk":
+            doc_id = row.get("document_id", "")
+            if not doc_id:
+                logger.debug(f"Row {i} missing document_id, skipping with_correct_chunk.")
+                continue
+
+            # Single-shot uses 'chunk_id', multi-hop uses 'source_chunk_ids'
+            if source_subset.startswith("single_shot"):
+                chunk_id = row.get("chunk_id", "")
+                chunk_text = doc_meta_map.get(doc_id, {}).get("chunks_map", {}).get(chunk_id, "")
+                doc_summary = doc_meta_map.get(doc_id, {}).get("document_summary", "")
+                user_prompt = GOLD_QA_USER_PROMPT.format(question=qtext, summary=doc_summary, document=chunk_text)
+                user_msg = {"role": "user", "content": user_prompt}
+                calls.append(InferenceCall(messages=[user_msg], tags=["with_correct_chunk"]))
+                row_map.append(i)
+
+            elif source_subset.startswith("multi_hop"):
+                chunk_ids = row.get("source_chunk_ids", [])
+                if not isinstance(chunk_ids, list):
+                    chunk_ids = []
+                doc_summary = doc_meta_map.get(doc_id, {}).get("document_summary", "")
+                combined_texts = []
+                for cid in chunk_ids:
+                    text_ = doc_meta_map.get(doc_id, {}).get("chunks_map", {}).get(cid, "")
+                    if text_:
+                        combined_texts.append(text_)
+                chunk_text_joined = "\n\n".join(combined_texts)
+
+                user_prompt = GOLD_QA_USER_PROMPT.format(
+                    question=qtext, summary=doc_summary, document=chunk_text_joined
+                )
+                user_msg = {"role": "user", "content": user_prompt}
+                calls.append(InferenceCall(messages=[user_msg], tags=["with_correct_chunk"]))
+                row_map.append(i)
+
+        else:
+            logger.warning(f"Unrecognized scenario '{scenario}' (row={i}), skipping.")
+            continue
+
+    return calls, row_map
+
+
+def _parse_and_assemble(
+    question_ds: Dataset, responses_dict: Dict[str, List[str]], row_map: List[int], scenario: str
+) -> Dataset:
+    """
+    For each model, parse <answer> from each response. We replicate the original row
+    and add columns: 'answering_model', 'answer', 'answer_fashion' = scenario.
+    """
+    final_records = {col: [] for col in question_ds.column_names}
+    final_records["answering_model"] = []
+    final_records["answer"] = []
+    final_records["answer_fashion"] = []
+
+    for model_name, model_responses in responses_dict.items():
+        logger.info(f"Processing {len(model_responses)} responses from model={model_name} for scenario={scenario}.")
+        n_common = min(len(model_responses), len(row_map))
```

Comment:
what is the role of n_common? (the name is not clear to me, please add one line of comment)

Reply:
added comment